### PR TITLE
Adding fingerprint tag

### LINKED ISSUE #35: Add (?<fingerprint>) data to new plaintextsecrets.json rules
@eddynaka

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #35

### CODE REVIEW

File: Src/Plugins/Security/SEC1001.SecurePlaintextSecrets.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
         },
         {
           "SubId": "GoogleApiKey",
-          "ContentsRegex": "AIza(?i)[0-9a-z-_]{35}",
+          "ContentsRegex": "(?<fingerprint>AIza(?i)[0-9a-z-_]{35})",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1001.SecurePlaintextSecrets.json:33]: >fingerprint [](start = 31, length = 11)

you actually don't need to add the fingerprint group when the entire match is the fingerprint: the tool will just use the primary match in any case. removing <fingerprint> here will result in one fewer regex match

File: Src/Plugins/Security/SEC1001.SecurePlaintextSecrets.json [hunk 2]
```diff
@@ -40,7 +40,7 @@
         },
         {
           "SubId": "SlackToken",
-          "ContentsRegex": "xox[pboa]-[0-9]{12}-[0-9]{12}-[0-9]{12}-[0-9a-z]{32}",
+          "ContentsRegex": "(?<fingerprint>xox[pboa]-[0-9]{12}-[0-9]{12}-[0-9]{12}-[0-9a-z]{32})",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1001.SecurePlaintextSecrets.json:43]: >fingerprint [](start = 31, length = 11)

same here, don't need this. i will now stop remarking on this condition, would be good to scrub for it.

File: Src/Plugins/Security/SEC1001.SecurePlaintextSecrets.json [hunk 3]
```diff
@@ -65,27 +65,27 @@
         },
         {
           "SubId": "AWSSecretAccessKey",
-          "ContentsRegex": "((\\\"|'|`)?((?i)aws)?_?((?i)secret)_?((?i)access)?_?((?i)key)?_?((?i)id)?(\\\"|'|`)?\\\\s{0,50}(:|=>|=)\\\\s{0,50}(\\\"|'|`)?[A-Za-z0-9/+=]{40}(\\\"|'|`)?)",
+          "ContentsRegex": "((\\\"|'|`)?((?i)aws)?_?((?i)secret)_?((?i)access)?_?((?i)key)?_?((?i)id)?(\\\"|'|`)?\\\\s{0,50}(:|=>|=)\\\\s{0,50}(\\\"|'|`)?0-9[0-9a-z/+=]{40}(\\\"|'|`)?)",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1001.SecurePlaintextSecrets.json:68]: >0-9 [](start = 154, length = 3)

did you meand to add this?

File: Src/Plugins/Security/SEC1001.SecurePlaintextSecrets.json [hunk 4]
```diff
@@ -65,27 +65,27 @@
...
           "MessageArguments": { "secretKind": "AWS Secret Access Key" }
         },
         {
           "SubId": "AWSSessionToken",
-          "ContentsRegex": "((\\\"|'|`)?((?i)aws)?_?((?i)session)?_?((?i)token)?(\\\"|'|`)?\\\\s{0,50}(:|=>|=)\\\\s{0,50}(\\\"|'|`)?[A-Za-z0-9/+=]{16,}(\\\"|'|`)?)",
+          "ContentsRegex": "((\\\"|'|`)?((?i)aws)?_?((?i)session)?_?((?i)token)?(\\\"|'|`)?\\\\s{0,50}(:|=>|=)\\\\s{0,50}(\\\"|'|`)?0-9[0-9a-z/+=]{16,}(\\\"|'|`)?)",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1001.SecurePlaintextSecrets.json:73]: >0-9 [](start = 132, length = 3)

same here, why did you add this?

**Comment by michaelcfanning** [comment]
A good start, we'll have more opportunity to refine once we get to more testing. Please remove all unnecessary fingerprint groups (for patterns where the complete match *is* the fingerprint, and then merge. nice job!

**Review by michaelcfanning — APPROVED** [review]
:shipit:

File: Src/Plugins/Security/SEC1001.SecurePlaintextSecrets.json (hunk 3 above)
Comment (eddynaka) [reply Src/Plugins/Security/SEC1001.SecurePlaintextSecrets.json:68]: its wrong...should be (?i)[0-9a-z]
